cmake_modules/IcebergThirdpartyToolchain.cmake

```diff
@@ -126,3 +126,68 @@ endfunction()
 if(ICEBERG_ARROW)
   resolve_arrow_dependency()
 endif()
+
+# ----------------------------------------------------------------------
+# Apache Avro
+
+function(resolve_avro_dependency)
+  prepare_fetchcontent()
+
+  set(AVRO_USE_BOOST
+      OFF
+      CACHE BOOL "" FORCE)
+
+  set(AVRO_BUILD_EXECUTABLES
+      OFF
+      CACHE BOOL "" FORCE)
+
+  set(AVRO_BUILD_TESTS
+      OFF
+      CACHE BOOL "" FORCE)
+
+  fetchcontent_declare(Avro
+                       ${FC_DECLARE_COMMON_OPTIONS}
+                       GIT_REPOSITORY https://github.com/apache/avro.git
+                       GIT_TAG 1144cb7322bab4cd1c8bf330a9c504a0d4252b56
+                       SOURCE_SUBDIR
+                       lang/c++
+                       FIND_PACKAGE_ARGS
+                       NAMES
+                       Avro
+                       CONFIG)
```

Comment:
I cloned your commits and tried to make it work on my end. However, it seems that the Avro C++ library does not export CMake config file: https://issues.apache.org/jira/browse/AVRO-3088. It means that we have to maintain a `FindAvro.cmake` file to manually find the header file and libraries from the installed location. I will try to fix this on the Avro side.

Reply:
That would be great, I had some issue with `fmt::fmt-header-only` target, which interface link of avrocpp.